#161255322 Pagination support for articles

njeri-ngigi opened the pull request:
#### What does this PR do?
- Adds a pagination component for articles
- Adds tests for this component
#### Description of tasks to be completed
> Pagination support for articles

**Why this is important**
- Add pagination to our articles to make article traversal easier.

#### Relevant PT stories
[Pagination support for articles](https://www.pivotaltracker.com/story/show/161255322)

#### Screenshots:
![image](https://user-images.githubusercontent.com/28973383/49812564-bfc30f00-fd76-11e8-89b2-81dc645e45df.png)
![image](https://user-images.githubusercontent.com/28973383/49812583-c3ef2c80-fd76-11e8-9172-33237f339681.png)

njeri-ngigi committed:
(Feature #161255322): Pagination
- Add pagination component
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Add pagination component
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Add action for fetching user stories
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Add action for fetching user stories
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Add pagination for user articles in the profile
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Add pagination for user articles in the profile
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Adjust page links css
[Feature #161255322]

njeri-ngigi committed:
(Feature #161255322): Pagination
- Adjust page links css
[Feature #161255322]

njeri-ngigi committed:
Merge branch 'develop' into ft-pagination-161255322

AnguleMathias committed:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- CRUD functionality for articles

[Feature #161255319]

AnguleMathias committed:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- tests for CRUD functionality for articles

[Feature #161255319]

AnguleMathias reviewed (changes requested):
Great work implementing the functionality but test coverage on the PR is 26.6% (50% is the threshold) thereby a negative deviation in relation to the develop branch. Work on improving the test coverage

Commit:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Refactor code

[Feature #161255319]

Commit:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Refactor code

[Feature #161255319]

AnguleMathias committed:
Merge branch 'ft-articles-161255319' of https://github.com/andela/ah-codeofduty-frontend into ft-articles-161255319

AnguleMathias committed:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- Modified edit article form
- Add article created date as a string
- Add article reading time
- Modified edit and delete button to appear only to article author

[Feature #161255319]

Commit:
(Feature #161255322): Pagination tests
- Add test to check for correct mounting of paginate component
[Feature #161255322]

Commit:
(Feature #161255322): Pagination tests
- Add test to check for correct mounting of paginate component
[Feature #161255322]

AnguleMathias committed:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- UI feedback implementation

[Feature #161255319]

AnguleMathias committed:
Feature(Articles CRUD): Users should be able to Create, Read, Delete and Update Articles

- UI feedback implementation

[Feature #161255319]

njeri-ngigi committed:
Merge pull request #19 from andela/ft-articles-161255319

#161255319 Users should be able to create articles

Commit:
(Feature #161255322): Pagination tests
- Add test to check for correct mounting of paginate component
[Feature #161255322]

Commit:
(Feature #161255322): Pagination
- Add pagination component to articles page
- Add pagination component to profile pages to show articles
- Adjust article reducer to have pageCount stored in state
[Feature #161255322]

maxwellgithinji reviewed (approved):
LGTM

AnguleMathias reviewed (approved):
Great work
